File: PakClassified.Client/src/app/Core/Model/Advertisment/Advertisement.model.ts
import { City } from "../location/city.model";
import { CityArea } from "../location/cityArea.model";
import { Role } from "../User/Role.model";
import { User } from "../User/User.model";
// import { Role } from "../Role/Role.model";
import { AdvertisementImage } from "./AdvertisementImage.model";
import { AdvertisementStatus } from "./AdvertisementStatus.model";
import { AdvertisementSubCategory } from "./AdvertisementSubCategory.model";
import { AdvertisementTag } from "./AdvertisementTag.model";
import { AdvertisementType } from "./AdvertisementType.model";

export interface AdvertisementModel {
    id: number;
    subCategoryId: number;
    subCategory?: AdvertisementSubCategory;
 
    statusId: number;
    status?: AdvertisementStatus;
 
    typeId: number;
    type?: AdvertisementType;
 
    postedById: number;
    postedBy?: User;
 
    cityAreaId: number;
    cityArea?: CityArea;
    city?: City;
 
    name: string;
    price: number;
    description: string;
    features: string
    hits: number;
    endsOn: string;
    startsOn: string;
    createdDate: string;
    image: string;
}

File: PakClassfied.Server/PakClassified.Models/PakClassified/AdvertisementModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using PakClassified.Models.Location;
using PakClassified.Models.User;

namespace PakClassified.Models.PakClassified
{
    public class AdvertisementModel : BaseDTO<int>
    {
        public int SubCategoryId { get; set; }
        public int StatusId { get; set; }
        public int TypeId { get; set; }
        public int PostedById { get; set; }
        public int CityAreaId { get; set; }
        public string Name { get; set; }
        public decimal Price { get; set; }
        public string? Description { get; set; }
        public string? Features { get; set; }
        public int Hits { get; set; }
        public DateTime EndsOn { get; set; }
        public string Image { get; set; }
        public DateTime StartsOn { get; set; }
        //public IEnumerable<AdvertisementImageModel> Images { get; set; }
        //public IEnumerable<AdvertisementTagsModel> Tags { get; set; }
        //public string CityArea { get; set; }
        //public CityModel City
        //{
        //    get
        //    {
        //        return CityArea.City;
        //    }
        //    set
        //    {
        //        CityArea.City = value;
        //    }
        //}
        //public AdvertisementStatusModel Status { get; set; }
        //public AdvertisementSubCategoryModel SubCategory { get; set; }
        //public AdvertisementTypeModel Type { get; set; }
        //public string PostedBy { get; set; }

        //public AdvertisementModel()
        //{
        //    Images = new List<AdvertisementImageModel>();
        //    Tags = new List<AdvertisementTagsModel>();
        //}
    }
}
